### PR TITLE
Add support for remote stt service and local tts

### PR DESCRIPTION
**Pull request by AIIX**
Adds support for sending raw audio from PCM to a mycroft-core instance located on a remote server via the messagebus socket.
- Currently timer based approach for recording and sending raw audio
- Added as gui app setting to be optionally available

### CODE REVIEW

**Comment by AIIX** [comment]
- Added support for cmake build option BUILD_LOCAL_TTS
- If enabled uses Qtexttospeech and speech dispatcher for TTS output

**Comment by notmart** [comment]
just some minor issues.
in the short term is ok.
in the long term, i wonder if mycroft-core could be modified/configured to send its actual utterance sound file on the main socket, so we can just playback that?

**Comment by AIIX** [comment]
> just some minor issues.
> in the short term is ok.
> in the long term, i wonder if mycroft-core could be modified/configured to send its actual utterance sound file on the main socket, so we can just playback that?

Haven't looked into what mimic/mimic-2 is sending the mycroft-audio client yet if it is retrievable as raw audio frames can try streaming it back on the socket and playing the audio in GUI, will look into it.

**Comment by AIIX** [comment]
Removed the previous qtexttospeech implementation and added option to stream the sound file produced via mimic from the main socket. This also removes the cmake option of BUILD_LOCAL_TTS and can be activated from a switch in GUI app.

File: import/audiorec.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      if (!info.isFormatSupported(format)) {
+         qWarning() << "Format is not supported.";
+         format = info.nearestFormat(format);
+     }
+
+    audio = new QAudioInput(format, this);
+    audio->start(&destinationFile);
+}
+
+void AudioRec::recordTStop(){
```
Comment (notmart) [inline import/audiorec.cpp]: minor nitpick: coding style,
{ on new line after functions

**Comment by AIIX** [comment]
Moved RemoteTTS and RemoteSTT under build flag "BUILD_REMOTE_TTS" and fixed previous coding style things

**Pull request merged**